c10/cuda/CUDACachingAllocator.h

```diff
@@ -125,9 +128,39 @@ struct SegmentInfo {
   int64_t active_size = 0;
   cudaStream_t stream = 0;
   bool is_large = false;
+  MempoolId_t owner_private_pool_id = {0, 0};
   std::vector<BlockInfo> blocks;
 };
 
+struct BlockState {
```

Comment:
A top level comment about this would be pretty helpful. In particular, it would be good to know what this has to be synchronized with (e.g., change one thing, change something else.)